packages/metal-quartz-collapse/src/MetalQuartzCollapse.js

```diff
@@ -1,15 +1,209 @@
+
+		this.shimUnsupportedTransition_(this.content);
+	}
+
+	/**
+	 * Adds CSS classes and properties to the `content` element when `collapsed`
+	 * is true
+	 */
+	close() {
+		dom.addClasses(this.content, this.closedClass);
+		dom.removeClasses(this.content, this.openClass);
+		dom.removeClasses(this.content, this.transitionClass);
+		this.content.setAttribute('aria-expanded', false);
+		this.content.style.removeProperty('height');
+	}
+
+	/**
+	 * Checks to see if browser supports CSS3 Transitions and returns the name 
+	 * of the transitionend event; returns false if it's not supported
+	 * @protected
+	 */
+	getTransitionEndEvent_() {
+		let el = document.createElement('metalQuartzTransitionEnd');
+
+		let transitionEndEvents = {
+			transition: 'transitionend',
+			WebkitTransition: 'webkitTransitionEnd',
+			MozTransition: 'transitionend',
+			OTransition: 'oTransitionEnd otransitionend'
+		};
+
+		for (let name in transitionEndEvents) {
+			if (el.style[name] !== undefined) {
+				return transitionEndEvents[name];
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Handles a `click` event on the headers.
+	 * @param {!Event} event
+	 * @protected
+	 */
+	handleClick_(event) {
+		this.toggle();
+	}
+
+	/**
+	 * Handles a `keydown` event on the headers.
+	 * @param {!Event} event
+	 * @protected
+	 */
+	handleKeydown_(event) {
+		if (event.keyCode === 13 || event.keyCode === 32) {
+			this.toggle();
+			event.preventDefault();
+		}
+	}
+
+	/**
+	 * Handles the `transitionend` event on the content.
+	 * @param {!Event} event
+	 * @protected
+	 */
+	handleTransitionEnd_(event) {
+		this.content.transitionType ? this.open() : this.close();
+
+		this.transitionEventHandler_.removeAllListeners();
+	}
+
+	/**
+	 * Adds CSS classes and properties to the `content` element when `collapsed`
+	 * is false
+	 */
+	open() {
+		dom.addClasses(this.content, this.closedClass);
+		dom.addClasses(this.content, this.openClass);
+		dom.removeClasses(this.content, this.transitionClass);
+		this.content.setAttribute('aria-expanded', true);
+		this.content.style.removeProperty('height');
+	}
+
+	/**
+	 * Fires a synthetic `transitionend` event for browsers that don't support
+	 * CSS3 transitions
+	 * @param {!Element} element
+	 * @protected
+	 */
+	shimUnsupportedTransition_(element) {
+		if (!this.supportsTransitionEnd) {
+			anim.emulateTransitionEnd(element);
+		}
+	}
+
+	/**
+	 * Syncs the `content` element according to the value of the `collapsed`
+	 * state, attaching and removing css properties and classes needed to open
+	 * and close the element.
+	 */
+	syncCollapsed() {
+		this.collapsed ? this.animateClose() : this.animateOpen();
+	}
+
+	/**
+	 * Syncs the component according to the value of the `headers` state,
+	 * attaching events to the new element and detaching from any previous one.
+	 */
+	syncHeaders() {
+		this.eventHandler_.removeAllListeners();
```

Comment:
You can pass in a css selector, a dom element, or an array of selectors and dom elements. It looks like Metal's `dom.on` function delegates it on the document for you if it's a css selector. I'm not sure if there's a way to delegate dom elements with Metal.